shroudstone/replay/parser.py

```diff
@@ -17,12 +17,48 @@
 logger = logging.getLogger(__name__)
 
 def decompress(replay: Union[Path, BinaryIO]) -> BytesIO:
-    """Decompress a gzipped stormgate replay, skipping the 16-byte header."""
+    """Decompress a gzipped stormgate replay with dynamic header size.
+    
+    The header contains its own size at offset 8, which allows for parsing
+    replays with different header sizes.
+    """
     if isinstance(replay, Path):
         replay = replay.open("rb")
     with replay:
-        replay.seek(16)
-        return BytesIO(gzip.decompress(replay.read()))
+        # Read the header size from the file itself (at offset 8)
+        replay.seek(8)
+        (header_size,) = struct.unpack("<i", replay.read(4))
+        
+        # As a safety check, ensure header_size is reasonable
+        if header_size < 16 or header_size > 256:
+            logger.warning(f"Unusual header size detected: {header_size}, falling back to 24")
+            header_size = 24  # Fall back to known working size
```

Comment:
If we're going to fall back to anything I think it should be 16, since this was the previous behaviour of shroudstone so it will ensure backwards compatibility. If we already know that all historical replays have the correct header size entry, I think it's best to follow the "fail fast" philosophy and raise an exception rather than trying to parse malformed data.